### PR TITLE
GH-34 emmit write counter as a metric

### PR DESCRIPTION
**Pull request by makslion**
Emmiting write counter as a metric and setting an upper limit for the counter after witch we give up

### CODE REVIEW

File: internal/controller/dnsrecord_controller.go [hunk 1]
```diff
@@ -336,8 +339,16 @@ func (r *DNSRecordReconciler) applyChanges(ctx context.Context, dnsRecord *v1alp
 		// generation has not changed but there are changes.
 		// implies that they were overridden - bump write counter
 		if !generationChanged(dnsRecord) {
-			dnsRecord.Status.WriteCounter++
-			logger.V(3).Info("Changes needed on the same generation of record")
+			if dnsRecord.Status.WriteCounter < WriteCounterLimit {
```
Comment (maleck13) [inline internal/controller/dnsrecord_controller.go:342]: do we set any dimension on this metric? Seems like we should set the the dnsRecord name and namespace as labels?

Comment (maleck13) [reply internal/controller/dnsrecord_controller.go:342]: As I read it we will increment this for all dns records so only ever 20 writes or am I mistaken

**Review by maleck13 — CHANGES_REQUESTED** [review]
I think the metric is not unique per dnsrecord?

**Comment by makslion** [comment]
image
This is how it will behave now. That should be much better

File: internal/controller/dnsrecord_controller.go (hunk 1 above)
Comment (makslion) [reply internal/controller/dnsrecord_controller.go:342]: yes, only 20 writes per generation of DNS record (frankly, if we have more then 2 writes per gen something is wrong with the DNS provider or, more accurately, something keeps overriding changes)

**Review by maleck13 — APPROVED** [review]

**Pull request merged**